lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -44,6 +48,47 @@ def __init__(self, config: PbenchServerConfig, logger: logging.Logger):
             role=API_OPERATION.READ,
         )
 
+    def get_paginated_obj(self, query: Query, json: JSON, url: str):
+        """
+        Helper function to return a paginated object containing slice of dataset
+        (constructed according to the page limit and a page number), next page url,
+        and total items count.
```

Comment:
This function signature is missing its return type -- which should be something like `Tuple[List, Dict[str, str]]` -- which would make the text description somewhat more clear (since the text implies that it returns a single object containing three things, when in fact it returns two things, a list and an object containing two things).